apko/private/apko_image.bzl

```diff
@@ -113,25 +116,27 @@ def apko_image(name, contents, config, tag, output = "oci", architecture = None,
      name:         of the target for the generated image.
      contents:     Label to the contents repository generated by translate_lock. See [apko-cache](./apko-cache.md) documentation.
      config:       Label to the `apko.yaml` file.
+     resolved:     Label to the `apko.resolved.json` file. Derived from the `config` file by default.
      output:       "oci" of  "docker",
      architecture: the CPU architecture which this image should be built to run on. See https://github.com/chainguard-dev/apko/blob/main/docs/apko_file.md#archs-top-level-element"),
      tag:          tag to apply to the resulting docker tarball. only applicable when `output` is `docker`
      args:         additional arguments to provide when running the `apko build` command.
      **kwargs:       other common arguments like: tags, visibility.
     """
+    config_label = native.package_relative_label(config)
+    effective_resolved = resolved if resolved else config_label.name.removesuffix(".yaml") + ".resolved.json"
     _apko_image(
         name = name,
         config = config,
+        resolved = effective_resolved,
         contents = contents,
         output = output,
         architecture = architecture,
-        tag = tag,
+        tag = tag if tag else name,
```

Comment:
this is not safe as two targets with same name may exist in two different packages.

Reply:
Reverted.